### PR TITLE
Added support for numeric nested fields type, isRequired, val and att…

### PR DESCRIPTION
**Pull request by lilHermit**
…ributes

    Updated tests too
See #11003

### CODE REVIEW

File: src/View/Form/ArrayContext.php [hunk 1]
```diff
@@ -177,7 +177,12 @@ public function val($field, $options = [])
             return null;
         }
 
-        return Hash::get($this->_context['defaults'], $field);
+        // Using Hash::check here incase the default value is actually null
+        if (Hash::check($this->_context['defaults'], $field)) {
+            return Hash::get($this->_context['defaults'], $field);
+        } else {
```
Comment (ADmad) [inline src/View/Form/ArrayContext.php]: `else` is not required since there is a `return` in `if` block.

File: src/View/Form/ArrayContext.php [hunk 2]
```diff
@@ -273,4 +288,17 @@ public function error($field)
 
         return Hash::get($this->_context['errors'], $field);
     }
+
+    /**
+     * Strips out any numeric nesting like users.0.age
+     *
```
Comment (ADmad) [inline src/View/Form/ArrayContext.php]: Please add an example of input and output for easier understanding.

**Review by ADmad — CHANGES_REQUESTED** [review]

**Comment by lilHermit** [comment]
Do you think the following it more efficient?!

```php
protected function stripNesting($field)
{
    return preg_replace('/\.\d/','',$field);
}
```

**Comment by markstory** [comment]
@lilHermit It does look much simpler.

**Comment by lilHermit** [comment]
@markstory definitely simpler but my concern was (and the reason I used implode/explode) keys starting with numeric will break for example `users.0.9tags` becomes `userstags`

I've have now tweaked the regex to fix this

**Pull request merged**